### ISSUE TITLE
Still working?

### ISSUE THREAD

**Issue by Boombrewer** [RaspberryPints/RaspberryPints#238]
Is anyone still working on this project? I was gonna start trying to hack my way through getting the flow sensors to provide data to the front end but didn't want to go through that pain if we're changing the foundation. It seems like all progress hung waiting on the switch to laravel.

**Comment by mcangeli** [comment]
Yes. It slowed a bit for me (I'm working on an auto install script) as I
lost my job and they took the laptop I was using. Got it back and now I
have free time so I'll be working on it more.

I also have basic flow meters working though it's glitchy and if there is
any kind of air or hiccup in the system it seriously throws things off.

**Comment by Boombrewer** [comment]
Sorry to hear about your job man. Are you working on the install for the conversion to MCV? I haven't jumped into figuring out MCV yet. But I have my flow meters hooked up and I'm willing to try my hand hacking them into some working condition.

**Comment by thadius856** [comment]
I'm in the pretty much the same boat as Boom.

**Comment by mcangeli** [comment]
I'm working on and updated installer for the v2 release. It has more
challenges as it has to edit the Apache config.

I'll post what I have for the flow meters (I had merged it previously but
it was rolled back until they got the mvc stuff worked out)

**Comment by Boombrewer** [comment]
So your script for the front end of the flow meters is for PHP not MCV?

**Comment by mcangeli** [comment]
The mc whatever is php.

It's based on the python script that kerber had. The python passes the info
to a php page which updates the database and refreshes the page.

**Comment by Boombrewer** [comment]
Ok. That's what I was guessing we'd have to do. I just wasn't sure if passing the info to a PHP page would be the same as passing it to the MCV (or whatever the new foundation is). Can you pass the us the python script when you get a chance (if you haven't upload it already)?

**Comment by Boombrewer** [comment]
I apologize for my ignorance. I thought MCV was a different language than PHP. No wonder I was confusing everyone.

**Comment by mcangeli** [comment]
It almost is...

But really it's just a framework that makes doing repetitive tasks easier.

**Comment by Boombrewer** [comment]
Yeah, from reading laravel.com, it sounded pretty nice. Thinking about downloading it for my new (spinoff) project for my fermenter. I want to do a combination of BrewPi and RPints. And I'd like to roll my temp control on my kegorator onto my RPi / Alamode setup. So many ideas & options...

**Comment by thadius856** [comment]
So... do we have a stable release with MVC to build off of? Does anybody know how to use it?

If not, it might be time to face the music. If that branch is dead, it may be wise to consider reverting to v1.0.3 and building from there.

**Comment by JayUnt** [comment]
The MVC conversion has been done for some time.  There just hasn't been tested by others to see if I missed anything.  There is a database conversion sql file that will convert all of the data to the new format.  There is no install/update sites like we had with the original release.  @mcangeli  talked about creating a bash script that the user can run on their pi that will pull down the latest code, run the sql scripts and make the apache config changes as needed.

But to do it manually this is what you need to do:

1) get the new code (i would move your existing code to a diff directory as a backup)
2) run the feature-flow_monitor.sql script.   (backup your db first)
3) In your apache config, change the directory of your site to the public folder within the new codebase.
4) go into /app/config/database.php and update the mysql array to have the appropriate database info.

I just recently ran through getting all of the code as a fresh install, and then executing the db scripts (schema, text_data, 184 - ALTER keg and feature-flow_monitor) and it ran fine on my pi.

**Comment by Boombrewer** [comment]
I ran into an issue installing my flow sensors but I'll try to get the code swapped out tonight and see how she runs. That will at least start moving this down the right path again. Hopefully I'll have my sensors online by this weekend.

**Comment by Boombrewer** [comment]
Should we delete everything in the /var/www folder before copying over all the new files?

I have my sensors hooked up and I'm working on updating all the software so I can start trying to break it.... I mean test it.

**Comment by Boombrewer** [comment]
For others who may be in my same boat....

2) run the feature-flow_monitor.sql script. (backup your db first)
From command line:
$sudo bash
cd /var/www/sql
mysql -u root raspberrypints < feature-flow_monitor.sql -p
**(you'll be prompted for a password which will be your mysql password)

@JayUnt if there's an easier way to do this, please let me know.  This is just how I managed to figure it out.  (at least I think it worked right... I'm still working through this)

@JayUnt Also, can you post some more details about steps 3 & 4?  I'm a little lost on 3 and 4 doesn't sounds like its going to be that fun either.  Thanks!

**Comment by JayUnt** [comment]
After you backup the original site files, delete them from the site. You shouldn't be them. But better safe than sorry.

3) My apache config is a bit different then what most are doing because I'm using virtual hosts. I'm doing this so I can have 2 rpints setup, one for my kegerator and one for development. I'm not home right now, but I can look tomorrow for they line to update.  

4) In the new code base there is an app folder, inside that there is a config folder and inside that is a database.php file. This is a the config file for the laraval site.  Goto the section that says 'mysql' => array(, and update the database/username/password values to match your database connection info.

**Comment by Boombrewer** [comment]
Step 3 must be pretty important because I've done everything else and I'm not having any luck getting anything (front side or back side).  What does the new address become?  I was assuming you pointed the browser at the public folder but I get an error when I do that.  But I'm assuming that's from not updating Apache as needed.

**Comment by JayUnt** [comment]
So I didn't run it, but this should be all you need to do.  

run `sudo nano /etc/apache2/sites-enabled/000-default`

Change `<Directory /var/www/>` to `<Directory /var/www/public/>`

run `sudo /etc/init.d/apache2 reload`

_http://www.noveldevices.co.uk/rp-apacheroot_

**Comment by Boombrewer** [comment]
Ok.  So I do all that and I get this when trying to access "my IP/index.php" remote or locally:
![image](https://cloud.githubusercontent.com/assets/6321429/3156367/3c9c6868-eadb-11e3-81c9-3d4fa791a6c0.png)

I made sure to reload after adjusting 000-default.  And by the way, you need to adjust two lines in case anyone else if trying to do this.

DocumentRoot /var/www    becomes   DocumentRoot /var/www/public
and I believe you have to change the line JayUnt mentioned as well but I'm not 100% sure since its still not working.

My folder permission for everything inside www is 755.  Do I need to change them to 777?

**Comment by JayUnt** [comment]
it looks like app/storage needs to be writable.  Try `sudo chmod -R app/storage`.  Im not sure why i didnt have to do this though.. weird.
_http://forumsarchive.laravel.io/viewtopic.php?id=6627_

**Comment by Boombrewer** [comment]
That did it.  Nice work.  A couple of points/issues from the first few minutes of checking it out are posted in a new issue.  Thanks for the help.

**Comment by JayUnt** [comment]
My pleasure.  I'm glad we are getting back on track with this. Thanks for kick starting it back into action.

**Comment by Skemp45** [comment]
I keep getting memory allotment issues, I upped my max memory too, I will have to try on another server
